ideascube/configuration/models.py

```diff
@@ -0,0 +1,76 @@
+from django.conf import settings
+from django.db import models
+
+from ideascube.models import JSONField
+
+from .errors import (
+    InvalidConfigurationValue,
+    InvalidStoredConfigurationValue,
+    NoSuchConfigurationKey,
+    NoSuchConfigurationNamespace,
+    )
+from .registry import REGISTRY
+
+
+class Configuration(models.Model):
+    class Meta:
+        ordering = ['-date']
+
+    namespace = models.CharField(max_length=40)
+    key = models.CharField(max_length=40)
+    value = JSONField()
+    actor = models.ForeignKey(settings.AUTH_USER_MODEL)
+    date = models.DateTimeField(auto_now=True)
+
+    @staticmethod
+    def _get_key_registry(namespace, key):
+        try:
+            namespace_registry = REGISTRY[namespace]
+
+        except KeyError:
+            raise NoSuchConfigurationNamespace(namespace)
+
+        try:
+            return namespace_registry[key]
+
+        except KeyError:
+            raise NoSuchConfigurationKey(namespace, key)
+
+    @classmethod
+    def get(cls, namespace, key):
+        key_registry = cls._get_key_registry(namespace, key)
+        expected_type = key_registry['python_type']
+
+        config = cls.objects.get(namespace=namespace, key=key)
+
+        if not isinstance(config.value, expected_type):
```

Comment:
Several things here:

- Indeed, if it is not suppose to happen, no need to check, but
- Early error detection is better that latter one
- I still disagree to return the default value if a value is present but with a wrong type. (Especially if the set should prevent a wrong type to be inserted in the base)

On top of that, I generally prefer to use a cast/check method instead of using isinstance.

    try:
        return expected_type(config.value)
    except:
        # Something get wrong during the cast
        raise InvalidStoredConfiguration(...)

For basic type, it is almost the same, it even allow to handle wrong type value stored (int(5) == int('5')). If what we care is the API, this is still valid (you will get a object of the type set in the REGISTERY)
Moreover, it allows the check of more complex type (List of string, python object build from dict).